mollie-payments-for-woocommerce/includes/mollie/wc/helper/settings.php

```diff
@@ -630,4 +618,94 @@ protected function warnWoo3xRequiredForKlarna( $content ) {
 		return $content;
 	}
 
+    /**
+     * Get current locale by WordPress
+     *
+     * Default to self::SETTING_LOCALE_DEFAULT_LANGUAGE
+     *
+     * @return string
+     */
+    protected function getCurrentLocale()
+    {
+        $locale = apply_filters(self::FILTER_WPML_CURRENT_LOCALE, get_locale());
+
+        // Convert known exceptions
+        $locale = $locale === 'nl_NL_formal' ? 'nl_NL' : $locale;
```

Comment:
This surely can work, but in the future maybe we could have a procedure to _normalize_ language codes? For example, remove everything after the 2nd part, or even simpler just trim the code to 5 characters.

Reply:
@XedinUnknown That's code that already existed, but I agree that we need to have a better way to handle edge cases.